conjure-java-core/src/test/java/com/palantir/conjure/java/parameterized/objects/GeneratorType.java

```diff
@@ -0,0 +1,342 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.conjure.java.parameterized.objects;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.errorprone.annotations.Immutable;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.Safe;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.processing.Generated;
+
+/**
+ * This class is used instead of a native enum to support unknown values. Rather than throw an exception, the
+ * {@link GeneratorType#valueOf} method defaults to a new instantiation of {@link GeneratorType} where
+ * {@link GeneratorType#get} will return {@link Value#UNKNOWN}.
+ *
+ * <p>For example, {@code GeneratorType.valueOf("corrupted value").get()} will return
+ * {@link Value#UNKNOWN}, but {@link GeneratorType#toString} will return "corrupted value".
+ *
+ * <p>There is no method to access all instantiations of this class, since they cannot be known at compile time.
+ */
+@SuppressWarnings("checkstyle:HiddenField")
+@Generated("com.palantir.conjure.java.types.EnumGenerator")
+@Safe
+@Immutable
+public final class GeneratorType {
```

Comment:
I don't think we're gaining anything by using checked-in conjure generated types for `FilePath` and `GeneratorType` -- I think we might as well use a standard java enum for `GeneratorType` and perhaps a standard `java.nio.file.Path` for the path, if possible